src/firstboot/firstboot.c

```diff
@@ -308,23 +308,27 @@ static int prompt_locale(void) {
         return 0;
 }
 
-static int process_locale(void) {
-        const char *etc_localeconf;
+static int process_locale(int rfd) {
+        _cleanup_free_ char *etc_localeconf = NULL;
         char* locales[3];
         unsigned i = 0;
         int r;
 
-        etc_localeconf = prefix_roota(arg_root, "/etc/locale.conf");
-        if (laccess(etc_localeconf, F_OK) >= 0 && !arg_force) {
-                log_debug("Found %s, assuming locale information has been configured.",
-                          etc_localeconf);
+        r = chase_symlinks_at(rfd, "etc/locale.conf",
```

Comment:
I think it would be better to chase the dir the file is in, instead of the file. i.e. use chase_symlinks() to pin the /etc/ dir in the image, and then use faccessat() on that dir with the locale.conf filename as argument.